.github/workflows/ci.yml

```diff
@@ -0,0 +1,332 @@
+  cross_compiling:
+    runs-on: ubuntu-20.04
+    needs: [rustfmt, minver, macos, linux_native_builds, rust_stable]
+    env:
+      BUILD: check
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          # Cross claims to support Android, but when it tries to run Nix's tests it
+          # reports undefined symbol references.
+          - target: aarch64-linux-android
+          - target: arm-linux-androideabi
+          - target: armv7-linux-androideabi
+          - target: i686-linux-android
+          - target: x86_64-linux-android
+          - target: arm-unknown-linux-musleabi
+          - target: x86_64-unknown-fuchsia
+          - target: x86_64-unknown-illumos
+          # Cross claims to support running tests on iOS, but it actually doesn't.
+          # https://github.com/rust-embedded/cross/issues/535
+          - target: aarch64-apple-ios 
+          # cargo hack tries to invoke the iphonesimulator SDK for iOS
+            NOHACK: true
+          # Cross claims to support Linux powerpc64, but it really doesn't.
+          # https://github.com/rust-embedded/cross/issues/441
+          - target: powerpc64-unknown-linux-gnu
+          - target: s390x-unknown-linux-gnu
+          - target: x86_64-unknown-linux-gnux32
+          - target: x86_64-unknown-netbsd
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: setup Rust
+        uses: dtolnay/rust-toolchain@master
+        with: 
+          toolchain: '${{ env.MSRV }}'
+          components: clippy
+
+      - name: install targets
+        run: rustup target add ${{ matrix.target }}
+
+      - name: build
+        uses: ./.github/actions/build
+        with:
+          TARGET: '${{ matrix.target }}'
+          BUILD: '${{ env.BUILD }}'
+          NOHACK: '${{ matrix.NOHACK  }}'
+
+      - name: before_cache_script
+        run: rm -rf $CARGO_HOME/registry/index
+  
+
+  redox:
+    runs-on: ubuntu-20.04
+    needs: [rustfmt, minver, macos, linux_native_builds, rust_stable]
+    env: 
+      TARGET: x86_64-unknown-redox
+      CLIPPYFLAGS: -D warnings
+      BUILD: check
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: setup Rust
+        # Redox's MSRV policy is unclear.  Until they define it, use nightly.
+        uses: dtolnay/rust-toolchain@nightly
+        with: 
+          components: clippy
+       
+      - name: install targets
+        run: rustup target add ${{ env.TARGET }}
+
+      - name: build
+        uses: ./.github/actions/build
+        with:
+          TARGET: '${{ env.TARGET }}'
+          BUILD: '${{ env.BUILD }}'
+          CLIPPYFLAGS: '${{ env.CLIPPYFLAGS }}'
+
+      - name: before_cache_script
+        run: rm -rf $CARGO_HOME/registry/index
+  
+
+
+  # Rust Tier 3 targets can't use Rustup
+  tier3:
+    runs-on: ubuntu-20.04
+    env: 
+      BUILD: check
+      ZFLAGS: -Zbuild-std
+      CLIPPYFLAGS: -D warnings
+    strategy:
+      fail-fast: true
```

Comment:
Again, I think we should set `fail-fast: false` here.  If the build gets this far, then any more failures will probably be OS-dependent.

Reply:
All the `fail-test`s are set to `false` now